master/internal/trial.go

```diff
@@ -152,7 +153,12 @@ func (t *trial) Receive(ctx *actor.Context) error {
 				Err: errors.Wrapf(msg.Error, "trial allocation failed"),
 			})
 		}
-
+	case *apiv1.KillTrialRequest:
```

Comment:
i'm not a huge fan of piping these api requests in like this. and it's possible if it is available it could be misused. how about a `PatchState` message with an informational 'reason' or 'cause' that gets piped through the whole flow (maybe elsewhere, too, like where preempt isn't necessarily the scheduler) and then just rely on the reason rather than guessing the intent of the caller (though it's a pretty good guess).